# TeloTales: Telomere Content Analysis Pipeline

This repository contains scripts and instructions for downloading BAM files from Phase 3 of the 1000 Genomes Project, analyzing telomeric content using qmotif, and saving output files with telomeric data. Each step in the pipeline is outlined below, along with requirements and execution instructions.

We utilized [NCBI’s 1000 Genomes Project server](https://www.ncbi.nlm.nih.gov/projects/faspftp/1000genomes/)


## Table of Contents
* Requirements
* File Descriptions
* Usage
* Detailed Steps

## Requirements
To run this pipeline, you'll need:
* **qmotif**: Download and install qmotif ([Documentation here](https://adamajava.readthedocs.io/en/latest/qmotif/qmotif_1_0/))
* **Java**: Required to run qmotif

Additional files to prepare:
* `files_list.txt`: Contains URLs of BAM and corresponding BAI index files from the 1000 Genomes Project
* `chrnames`: Contains chromosome names for tallying telomeric reads

## File Descriptions
* **download_files.sh**: Bash script to download BAM and BAI files from Phase 3 of the 1000 Genomes Project using `files_list.txt`

* **runqmotif.py**: Python script to run qmotif on BAM files to analyze telomeric content
* **stage2.py**: Python script to parse `log.txt` files generated by runqmotif.py to extract telomere read counts by chromosome
* **realcoverage.sh**: Bash script to tally telomere reads by chromosome; uses `chrnames` file
* **scaledgenomic.sh**: Bash script to parse `output.txt` files generated by runqmotif.py to extract total telomeric reads per sample

## Usage
## Sequence of Execution
To use the pipeline, run the scripts in the following order:
1. `download_files.sh` (requires: `files_list.txt`)
2. `runqmotif.py` (generates: `log.txt` and `output.txt` files)
3. `stage2.py` (requires: `log.txt` files generated by `runqmotif.py`)
4. `realcoverage.sh` (requires: `chrnames` file; generates: `output_coverage_filenames`, `stage2coverage`)
5. `scaledgenomic.sh` (requires: `output.txt` files generated by `runqmotif.py`)


## Detailed Steps
1. **Download BAM and BAI files**
* Run `download_files.sh` to download the BAM and BAI files from NCBI’s 1000 Genomes Project server. Make sure to:
* Include URLs for BAM and BAI files in `files_list.txt` and that both `download_files.sh` and `files_list.txt` are in the same folder.

```
chmod +x download_files.sh
```

```
./download_files.sh
```
**OR**

```
bash download_files.sh
```

2. **Run qmotif with runqmotif.py**
* Before running runqmotif.py, ensure qmotif is installed, and the path to qmotif and your BAM and BAI input files is set.

```
python3 runqmotif.py
```








3.	stage2.py requires the log files generated by hope2_bblah.py (check what working directory you’re in before proceeding.) This script will give you the chromosome number and the final telomere motif counts for each chromosome after stage 2 -> this output will be stored in “{sequence_name}_stage2_coverage.txt file
4.	coverage.sh is to get a tally of telomere motifs for each chromosome. This script uses “chrnames” file, make sure they’re in the same folder. This script creates only one output file “stage2coverage” which has all the info on all the files you ran. 
5.	totaltelomere.py gives total telomere motif counts for the sequence across all chromosomes. This uses the “stage2coverage” file we got as an output in the previous step and generates a file named “total_telomere_motif_counts”


