---
title: "across_chromosomes"
author: "Priyanshi Shah"
date: "2024-08-28"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(dplyr)
library(colorspace)
```

## R Markdown


```{r loadingdata}
final_across <- read.csv("chapter1_final_across_chromosomes.csv", header = TRUE, row.names = NULL)
head(final_across)
View(final_across) #Just checking out the data
summary(final_across)

```


```{r EliminatingMissingData}
#final_across <- na.omit(final_across)

final_across <- final_across %>% filter(!is.na(rawGenomicCount))
#View(final_across)
```


```{r DataTypeConversion}
# Converting columns to appropriate data types

final_across$rawGenomicCount <- as.numeric(final_across$rawGenomicCount)
```



```{r Analysis}

# Extract Telomere motif counts for all the Populations 

acb_pop <- final_across %>% filter(population == "ACB")
asw_pop <- final_across %>% filter(population == "ASW")
beb_pop <- final_across %>% filter(population == "BEB")
cdx_pop <- final_across %>% filter(population == "CDX")
ceu_pop <- final_across %>% filter(population == "CEU")
chb_pop <- final_across %>% filter(population == "CHB")
chs_pop <- final_across %>% filter(population == "CHS")
clm_pop <- final_across %>% filter(population == "CLM")
esn_pop <- final_across %>% filter(population == "ESN")
fin_pop <- final_across %>% filter(population == "FIN")
gbr_pop <- final_across %>% filter(population == "GBR")
gih_pop <- final_across %>% filter(population == "GIH")
gwd_pop <- final_across %>% filter(population == "GWD")
ibs_pop <- final_across %>% filter(population == "IBS")
itu_pop <- final_across %>% filter(population == "ITU")
jpt_pop <- final_across %>% filter(population == "JPT")
khv_pop <- final_across %>% filter(population == "KHV")
lwk_pop <- final_across %>% filter(population == "LWK")
msl_pop <- final_across %>% filter(population == "MSL")
mxl_pop <- final_across %>% filter(population == "MXL")
pel_pop <- final_across %>% filter(population == "PEL")
pjl_pop <- final_across %>% filter(population == "PJL")
pur_pop <- final_across %>% filter(population == "PUR")
stu_pop <- final_across %>% filter(population == "STU")
tsi_pop <- final_across %>% filter(population == "TSI")
yri_pop <- final_across %>% filter(population == "YRI")

#View(chs_pop)
#View(pur_pop)
```


```{r SettingColorPalette}
my_palette <- qualitative_hcl(22, palette = "Dark 3")  

print(length(my_palette))
```



```{r BoxPlot for ACB pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the ACB population and chromosomes 1 to 22
filtered_acb_data <- acb_pop %>%
  filter(population == "ACB", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_acb_counts <- filtered_acb_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
acb <- ggplot(filtered_acb_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the African Caribbean Population (ACB) in Barbados ",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0,2000)+
   theme(
    axis.text.x = element_text(size = 10, face = "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("ACB_autosomewise.png", plot = acb, width = 10, height = 8, dpi = 300)


```


```{r BoxPlot for ASW pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the ASW population and chromosomes 1 to 22
filtered_asw_data <- asw_pop %>%
  filter(population == "ASW", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_asw_counts <- filtered_asw_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
asw <- ggplot(filtered_asw_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the African Population (ASW) in Southwest US ",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0,1000)+
   theme(
    axis.text.x = element_text(size = 10, face = "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("ASW_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)

```





```{r BoxPlot for BEB pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the BEB population and chromosomes 1 to 22
filtered_beb_data <- beb_pop %>%
  filter(population == "BEB", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_beb_counts <- filtered_beb_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
beb <- ggplot(filtered_beb_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Bengali in Bangladesh (BEB) Population",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0,1000)+
   theme(
    axis.text.x = element_text(size = 10, face = "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("BEB_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)

```


```{r BoxPlot for CDX pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the CDX population and chromosomes 1 to 22
filtered_cdx_data <- cdx_pop %>%
  filter(population == "CDX", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_cdx_counts <- filtered_cdx_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
cdx  <- ggplot(filtered_cdx_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Chinese Dai Population (CDX) in Xishuangbanna, China",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 1000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("CDX_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)

```


```{r BoxPlot for CEU pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the CEU population and chromosomes 1 to 22
filtered_ceu_data <- ceu_pop %>%
  filter(population == "CEU", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_ceu_counts <- filtered_ceu_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
ceu  <- ggplot(filtered_ceu_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in Utah residents with Northern and Western European ancestry (CEU) in Utah",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0,1200)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("CEU_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)

```


```{r BoxPlot for CHB pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the CHB population and chromosomes 1 to 22
filtered_chb_data <- chb_pop %>%
  filter(population == "CHB", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_chb_counts <- filtered_chb_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
chb  <- ggplot(filtered_chb_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Han Chinese Population (CHB) in Beijing, China ",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0,2500)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("CHB_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```





```{r BoxPlot for CHS pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the CHS population and chromosomes 1 to 22
filtered_chs_data <- chs_pop %>%
  filter(population == "CHS", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length  per chromosome
avg_chs_counts <- filtered_chs_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
chs  <- ggplot(filtered_chs_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  scale_fill_manual(values = my_palette, guide = "none")+
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Han Chinese South (CHS) Population",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  ylim(0.0,2500)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("CHS_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```



```{r BoxPlot for CLM pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the CLM population and chromosomes 1 to 22
filtered_clm_data <- clm_pop %>%
  filter(population == "CLM", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_clm_counts <- filtered_clm_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
clm  <- ggplot(filtered_clm_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Colombian Population (CLM) in Medellin, Colombia",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 6500)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("CLM_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```



```{r BoxPlot for ESN pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the ESN population and chromosomes 1 to 22
filtered_esn_data <- esn_pop %>%
  filter(population == "ESN", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_esn_counts <- filtered_esn_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
esn  <- ggplot(filtered_esn_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Esan Population (ESN) in Nigeria",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim (0.0, 4100)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("ESN_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```



```{r BoxPlot for FIN Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the FIN population and chromosomes 1 to 22
filtered_fin_data <- fin_pop %>%
  filter(population == "FIN", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_fin_counts <- filtered_fin_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
fin  <- ggplot(filtered_fin_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Finnish (FIN) Population in Finland",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim (0.0, 3000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("FIN_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)




```




```{r BoxPlot for GBR Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the GBR population and chromosomes 1 to 22
filtered_gbr_data <- gbr_pop %>%
  filter(population == "GBR", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_gbr_counts <- filtered_gbr_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
gbr  <- ggplot(filtered_gbr_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the British (GBR) Population",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 2000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("GBR_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)




```


```{r BoxPlot for GIH Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the GIH population and chromosomes 1 to 22
filtered_gih_data <- gih_pop %>%
  filter(population == "GIH", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_gih_counts <- filtered_gih_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
gih  <- ggplot(filtered_gih_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Gujarati Indians in Houston, Texas Population (GIH)",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 500)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("GIH_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)




```


```{r BoxPlot for GWD Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the GWD population and chromosomes 1 to 22
filtered_gwd_data <- gwd_pop %>%
  filter(population == "GWD", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_gwd_counts <- filtered_gwd_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
gwd  <- ggplot(filtered_gwd_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Gambian Population (GWD) in Western Division, The Gambia",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim (0.0, 3000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("GWD_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```



```{r BoxPlot for IBS Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the IBS population and chromosomes 1 to 22
filtered_ibs_data <- ibs_pop %>%
  filter(population == "IBS", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_ibs_counts <- filtered_ibs_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
ibs  <- ggplot(filtered_ibs_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Iberian Population (IBS) in Spain",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 800)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("IBS_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```




```{r BoxPlot for ITU pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the ITU population and chromosomes 1 to 22
filtered_itu_data <- itu_pop %>%
  filter(population == "ITU", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length  per chromosome
avg_itu_counts <- filtered_itu_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
itu  <- ggplot(filtered_itu_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  scale_fill_manual(values = my_palette, guide = "none")+
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Indian Telugu Population (ITU) in the UK",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  ylim(0.0,2000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("ITU_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```



```{r BoxPlot for JPT pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the JPT population and chromosomes 1 to 22
filtered_jpt_data <- jpt_pop %>%
  filter(population == "JPT", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_jpt_counts <- filtered_jpt_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
jpt  <- ggplot(filtered_jpt_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Japanese Population in Tokyo (JPT), Japan ",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0,3500)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("JPT_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```




```{r BoxPlot for KHV Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the KHV population and chromosomes 1 to 22
filtered_khv_data <- khv_pop %>%
  filter(population == "KHV", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_khv_counts <- filtered_khv_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
khv  <- ggplot(filtered_khv_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Kinh Population (KHV) in Ho Chi Minh City, Vietnam",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 500)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("KHV_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```


```{r BoxPlot for LWK pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the LWK population and chromosomes 1 to 22
filtered_lwk_data <- lwk_pop %>%
  filter(population == "LWK", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_lwk_counts <- filtered_lwk_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
lwk  <- ggplot(filtered_lwk_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Luhya Population in Webuye (LWK), Kenya ",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 5000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("LWK_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```





```{r BoxPlot for MSL Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the MSL population and chromosomes 1 to 22
filtered_msl_data <- msl_pop %>%
  filter(population == "MSL", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_msl_counts <- filtered_msl_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
msl  <- ggplot(filtered_msl_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Mende Population (MSL) in Sierra Leone",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 3000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("MSL_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)

```



```{r BoxPlot for MXL pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the MXL population and chromosomes 1 to 22
filtered_mxl_data <- mxl_pop %>%
  filter(population == "MXL", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_mxl_counts <- filtered_mxl_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
mxl  <- ggplot(filtered_jpt_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Mexican Population in Los Angeles (MXL) in California",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0,4000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("MXL_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```







```{r BoxPlot for PEL Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the PEL population and chromosomes 1 to 22
filtered_pel_data <- pel_pop %>%
  filter(population == "PEL", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_pel_counts <- filtered_pel_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
pel  <- ggplot(filtered_pel_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Peruvian Population (PEL) in Lima, Peru",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 2000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("PEL_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```



```{r BoxPlot for PJL Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the PJL population and chromosomes 1 to 22
filtered_pjl_data <- pjl_pop %>%
  filter(population == "PJL", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_pjl_counts <- filtered_pjl_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
pjl <- ggplot(filtered_pjl_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Punjabi Population (PJL) in Lahore, Pakistan",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 1000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("PJL_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```



```{r BoxPlot for PUR pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the PUR population and chromosomes 1 to 22
filtered_pur_data <- pur_pop %>%
  filter(population == "PUR", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_pur_counts <- filtered_pur_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
 pur <- ggplot(filtered_pur_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Puerto Rican (PUR) Population",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim(0.0, 1500)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("PUR_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```



```{r BoxPlot for STU Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the STU population and chromosomes 1 to 22
filtered_stu_data <- stu_pop %>%
  filter(population == "STU", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_stu_counts <- filtered_stu_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
stu <- ggplot(filtered_stu_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Sri Lankan Tamil (STU) Population in the UK",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim (0.0, 1500)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
   plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("STU_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```


```{r BoxPlot for TSI Pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the TSI population and chromosomes 1 to 22
filtered_tsi_data <- tsi_pop %>%
  filter(population == "TSI", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length per chromosome
avg_tsi_counts <- filtered_tsi_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
tsi <- ggplot(filtered_tsi_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Toscani Population (TSI) in Italy",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  scale_fill_manual(values = my_palette, guide = "none")+
  ylim (0.0, 2000)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
   plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("TSI_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```




```{r BoxPlot for YRI pop}
# Necessary libraries already loaded at the beginning

# Filtering the data for the YRI population and chromosomes 1 to 22
filtered_yri_data <- yri_pop %>%
  filter(population == "YRI", chromosome %in% paste0("chr", 1:22))

# Calculate the average telomere length  per chromosome
avg_yri_counts <- filtered_yri_data %>%
  group_by(chromosome) %>%
  summarise(mean_raw_genomic_reads = mean(rawGenomicCount))

# Create the boxplot
yri <- ggplot(filtered_yri_data, aes(x = chromosome, y = rawGenomicCount, fill = chromosome)) +
  geom_boxplot() +
  scale_fill_manual(values = my_palette, guide = "none")+
  labs(title = "Autosome-Wise Assessment of Telomere Content in the Yoruba Population (YRI) in Ibadan, Nigeria",
       x = "Autosomes",
       y = "Average Telomere Content", face = "bold", size = 18) +
  theme_minimal() +
  scale_x_discrete(limits = paste0("chr", 1:22)) +
  ylim(0.0,1500)+
   theme(
    axis.text.x = element_text(size = 10, face= "bold"),  # Adjust the font size for x-axis labels
    axis.text.y = element_text(size = 16),  # Adjust the font size for y-axis labels
    plot.background = element_rect(fill = "white", color = NA),
    panel.background = element_rect(fill = "white", color = NA)
  )

ggsave("YRI_autosomewise.png", plot = , width = 10, height = 8, dpi = 300)


```








